app/api/sessions.py

```diff
@@ -104,12 +106,24 @@ def validate_date(self, data, original_data):
         type_='speaker')
 
 
-class SessionList(ResourceList):
+class SessionListPost(ResourceList):
     """
-    List and create Sessions
+    List Sessions
     """
     def before_post(self, args, kwargs, data):
         require_relationship(['event'], data)
+        if not has_access('is_coorganizer', event_id=data['event']):
```

Comment:
@shubham-padia just a note. 400 isn't only for a malformed syntax. Any client side mistake in the data that the server cannot process. It can be a syntax error, invalid request or even a deceptive request according to RFC 7231. So you can use 422 Unprocessable entity also, but 400 Bad Request also covers unprocessable stuff.  It's more like 422 is used when the request is valid and processable but you can't or don't want to process the particular instruction. 400 you use when it is either syntactically wrong, or syntactically correct but invalid for your system. Though according to RFC 2616, whatever you said was true.

Just wanted to notify.